common/chroot-style/ethereal.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/sh
+#
+# This chroot script uses symlinks to emulate being in a chroot,
+# it also symlinks the root directory to the masterdir and assumes
+# base-chroot is installed in the host system.
+#
+# It will damage your host system, only use it in disposable
+# containers.
+#
+# Note that this assumes the following.
+# The xbps-src git checkout not is mounted at /void-packages
+# There is a symlink from / to /void-packages/masterdir
+#
+
+readonly MASTERDIR="$1"
+readonly DISTDIR="$2"
+readonly HOSTDIR="$3"
+readonly EXTRA_ARGS="$4"
+readonly CMD="$5"
+shift 5
+
+if [ -z "$MASTERDIR" -o -z "$DISTDIR" ]; then
+	echo "$0 MASTERDIR/DISTDIR not set"
+	exit 1
+fi
+
+msg_red() {
+	# error messages in bold/red
+	[ -n "$NOCOLORS" ] || printf >&2 "\033[1m\033[31m"
+	printf "=> ERROR: %s\\n" "$@"
+	[ -n "$NOCOLORS" ] || printf >&2 "\033[m"
+}
+
+fake_mount() {
+	if [ -d "$2" ] && ! rmdir "$2" >/dev/null 2>&1; then
+		msg_red "Failed to remove $2, not empty ?\n"
+		exit 1
+	fi
+
+	[ -f "$2" -o -L "$2" ] && rm -f "$2"
+
+	ln -s "$1" "$2"
+	echo "linked $1 -> $2"
+}
+
+if [ "${XBPS_ALLOW_CHROOT_BREAKOUT}" != "yes" ]; then
+	msg_red "chroot-style 'ethereal' requires XBPS_ALLOW_CHROOT_BREAKOUT=yes\n"
+	msg_red "This chroot-style is meant for disposable containers and will destroy your system\n"
+	exit 1
+fi
+
+if [ ! -L "$MASTERDIR" -o "$(readlink "$MASTERDIR")" != "/" ]; then
+	msg_red "$MASTERDIR isn't symlinked to /!\n"
+	exit 1
+fi
+
+fake_mount "$DISTDIR" "$MASTERDIR"/void-packages
+
+# Do the same for hostdir
+if [ -n "$HOSTDIR" ]; then
+	fake_mount "$HOSTDIR" "$MASTERDIR"/host
+fi
+
+# xbps-src may send some other binds, parse them here
+while getopts 'b:' c -- "$EXTRA_ARGS"; do
+	# Skip everything that's not a bind
+	[ "$c" = "b" ] || continue
+
+	from="${OPTARG%:*}"
+	to="${OPTARG#*:}"
+
+	fake_mount "$from" "$to"
+
+	mounts="${mounts} $to"
+done
+
+# Store current directory for returning later
+OLDPWD="$(pwd)"
+
+# To give the illusion we entered the chroot, cd to /
+cd /
```

Comment:
This should probably have an || error case to handle the unlikely event that the cd fails.